webapp/client/client.jsx

```diff
@@ -1767,6 +1769,30 @@ export default class Client {
             end(this.handleResponse.bind(this, 'getFileInfosForPost', success, error));
     }
 
+    getOpenGraphMetaData(url, success, error) {
+        // There can be only one on going request for one url.
+        // This helps where there are multiple posts with same URL are rendered simultaneously.
+        var self = this;
+        if (!this.openGraphMetaDataOnGoingRequests[url]) {
```

Comment:
@enahum, Currently the this function is only called from https://github.com/mattermost/platform/pull/4890/files/68587e43d1b4e38a9f47ee69994b36fcb217b875#diff-f1f7963b47ead3e8ded08b371f14420fR600. And in case of simultaneous requests for same URL, success function of only one request would be called. Which is fine as the success function triggers and action to update OG data, which update update it for all posts and they would be rendered with data received from the URL.

Though I understand that as this is a client function it can be called with success/error functions, in which case only the first request's success/error function would be called. I can move logic of rejecting more than one request with same URL to 'updateOpenGraphMetaData' action instead.